sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -0,0 +1,208 @@
+
+            public ObjectDisposedPointerCheck(ExplodedGraph explodedGraph)
+                : base(explodedGraph)
+            {
+            }
+
+            public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                var instruction = programPoint.Block.Instructions[programPoint.Offset] as InvocationExpressionSyntax;
+
+                return instruction == null
+                    ? programState
+                    : VisitInvocationExpression(instruction, programState);
+            }
+
+            public override ProgramState PreProcessUsingStatement(ProgramPoint programPoint, ProgramState programState)
+            {
+                var newProgramState = programState;
+
+                var usingFinalizer = (UsingFinalizerBlock)programPoint.Block;
+
+                var disposables = usingFinalizer.Disposables
+                    .Select(disposable =>
+                    new
+                    {
+                        Disposable = disposable,
+                        Symbol = semanticModel.GetDeclaredSymbol(disposable)
+                            ?? semanticModel.GetSymbolInfo(disposable).Symbol
+                    });
+
+                foreach (var nodeWithSymbol in disposables)
+                {
+                    newProgramState = ProcessDisposableSymbol(newProgramState, nodeWithSymbol.Disposable, nodeWithSymbol.Symbol);
+                }
+
+                newProgramState = ProcessStreamDisposingTypes(newProgramState, (SyntaxNode)usingFinalizer.UsingStatement.Expression ?? usingFinalizer.UsingStatement.Declaration);
+
+                return newProgramState;
+            }
+
+            private ProgramState ProcessStreamDisposingTypes(ProgramState programState, SyntaxNode usingExpression)
+            {
+                var newProgramState = programState;
+
+                var arguments = usingExpression.DescendantNodes()
+                    .OfType<ObjectCreationExpressionSyntax>()
+                    .Where(this.IsStreamDisposingType)
+                    .Select(FirstArgumentOrDefault)
+                    .WhereNotNull();
+
+                foreach (var argument in arguments)
+                {
+                    var streamSymbol = semanticModel.GetSymbolInfo(argument.Expression).Symbol;
+                    newProgramState = ProcessDisposableSymbol(newProgramState, argument.Expression, streamSymbol);
+                }
+
+                return newProgramState;
+            }
+
+            private static ArgumentSyntax FirstArgumentOrDefault(ObjectCreationExpressionSyntax objectCreation) =>
+                objectCreation.ArgumentList?.Arguments.FirstOrDefault();
+
+            private bool IsStreamDisposingType(ObjectCreationExpressionSyntax objectCreation) =>
+                semanticModel.GetSymbolInfo(objectCreation.Type)
+                    .Symbol
+                    .GetSymbolType()

```

Comment:
Is it ok to call `.GetSymbolType()` on null? Are we sure `.Symbol` is never null?